src/target/target_kind.cc

```diff
@@ -143,6 +143,33 @@ void CheckOrSetAttr(Map<String, ObjectRef>* attrs, const String& name, const Str
 
 /**********  Target kind attribute updaters  **********/
 
+/*!
+ * \brief Update the attributes in the CUDA target.
+ * \param attrs The original attributes
+ * \return The updated attributes
+ */
+Map<String, ObjectRef> UpdateCUDAAttrs(Map<String, ObjectRef> attrs) {
+  // Update -arch=sm_xx
+  int archInt;
+  if (attrs.count("arch")) {
+    // If -arch has been specified, validate the correctness
+    String archStr = Downcast<String>(attrs.at("arch"));
+    archInt = ExtractIntWithPrefix(archStr, "sm_");
+    ICHECK(archInt != -1) << "ValueError: CUDA target gets an invalid CUDA arch: -arch=" << archStr;
+  } else {
+    // Use the compute version of the first CUDA GPU instead
+    TVMRetValue version;
+    if (!DetectDeviceFlag({kDLCUDA, 0}, runtime::kComputeVersion, &version)) {
+      LOG(WARNING) << "Unable to detect CUDA version, default to \"-arch=sm_20\" instead";
+      archInt = 20;
```

Comment:
Our of curiosity, why sm_20?

Reply:
1. To keep consistent with the `LLVM NVPTX target`, https://github.com/apache/tvm/blob/4bb6f3133ef7122543a861ea7d2610515e8e9a68/src/target/target_kind.cc#L164

2. `sm_20` is the lowerest CUDA arch, and is no longer be supported by most of NV devices. When the code runs here, an error will be reported. This will be an another way to remind users to specify cuda arch manually. lol

Reply:
I see...Although I feel the default should be a more reasonable `arch`, it's also fine to follow the existing one.